lsp/src/main/kotlin/org/gradle/declarative/lsp/DeclarativeTextDocumentService.kt

```diff
@@ -300,7 +300,8 @@ class DeclarativeTextDocumentService : TextDocumentService {
         val fileName = uri.path.substringAfterLast('/')
         val fileSchema = schemaAnalysisEvaluator.evaluate(fileName, text)
         val settingsSchema = schemaAnalysisEvaluator.evaluate(
-            declarativeResources.settingsFile.name, declarativeResources.settingsFile.readText()
+            declarativeResources.settingsFile.name, 
+            declarativeResources.settingsFile.takeIf { it.canRead() }?.readText().orEmpty()
```

Comment:
🤔 Now I realize we don't handle these cases when adding a new settings file.

Reply:
I decided to merge this as-is, so we have a (partially, maybe) working build sooner. Let's maybe track the problem you found as an issue?